test/functional/feature_bind_port_discover.py

```diff
@@ -91,10 +91,10 @@ def run_test(self):
                     found_addr2 = True
                     assert_equal(local['port'], self.bind_ports[i])
             if not found_addr1:
-                self.log.error(f"Address {ADDR1} not found in node{i}'s local addresses: {local}")
+                self.log.error(f"Address {ADDR1} not found in node{i}'s local addresses: {self.nodes[i].getnetworkinfo()['localaddresses']}")
                 assert False
             if not found_addr2:
-                self.log.error(f"Address {ADDR2} not found in node{i}'s local addresses: {local}")
+                self.log.error(f"Address {ADDR2} not found in node{i}'s local addresses: {self.nodes[i].getnetworkinfo()['localaddresses']}")
```

Comment:
If you really want to log the RPC output, you could capture it in a variable just before the `for local in...` and then iterate that variable inside the for loop, and print it in the error message. It's easy to reproduce this if you need to test it (I had trouble getting 1.1.1.1 and 2.2.2.2 set up on my machine for this test!)